x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/RuleQueryBuilder.java

```diff
@@ -104,16 +110,22 @@ private RuleQueryBuilder(
         if (matchCriteria == null || matchCriteria.isEmpty()) {
             throw new IllegalArgumentException("matchCriteria must not be null or empty");
         }
-        if (Strings.isNullOrEmpty(rulesetId)) {
-            throw new IllegalArgumentException("rulesetId must not be null or empty");
+        if (rulesetIds == null || rulesetIds.isEmpty()) {
+            throw new IllegalArgumentException("rulesetIds must not be null or empty");
+        }
+
+        if (rulesetIds.size() > MAX_NUM_RULESETS) {
```

Comment:
completely unrelated to your changes, I do wonder if there are other places we make these types of validations, like maximum number of elements, checking for empty arrays or for `null` values in arrays - maybe these could be part of `PARSER.declareStringArray`? WDYT?

Reply:
I do like having the Exceptions in the constructor for testing purposes, but we can certainly look at standardizing these errors more. I'd like to keep that in a separate tech debt/cleanup initiative if you're OK with that though.